### ISSUE TITLE
VariableDeclarator cannot accommodate multiline const declarations with 4-space indent

### ISSUE THREAD

**Issue by jonathanj** [eslint/eslint#7042]
**What version of ESLint are you using?**

3.4.0

**What parser (default, Babel-ESLint, etc.) are you using?**

Babel-ESLint

**Please show your full configuration:**

```
{
  "extends": "eslint:recommended",
  "env": {
    "browser": true
  },
  "plugins": [
    "react"
  ],
  "rules": {
    "linebreak-style": [
      "error",
      "unix"
    ],
    "quotes": [
      "warn",
      "single"
    ],
    "indent": [
      "error",
      4,
      {
        "SwitchCase": 1,
        "VariableDeclarator": {
          "var": 1,
          "let": 1,
          "const": 1
        }
      }
    ],
    "semi": [
      "error",
      "never"
    ],
    "no-unused-vars": [
      "error",
      {
        "args": "none"
      }
    ]
  },
  "parser": "babel-eslint",
  "parserOptions": {
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true,
      "experimentalObjectRestSpread": true
    }
  },
  "ecmaVersion": 6
}
```

**What did you do? Please include the actual source code causing the issue.**

```javascript
function foo() {
    const a = 1,
          b = 2
}
```

**What did you expect to happen?**

I expected no linting errors regarding the indentation of the declaration of `b`.

**What actually happened? Please include the actual, raw output from ESLint.**

```
  3:11  error  Expected indentation of 8 space characters but found 10  indent
```

Since `VariableDeclarator` specifies indentations as multiples of the indent setting (4) and `const<space>` is not a multiple of 4 characters long and `VariableDeclarator` doesn't allow fractional indentations, it's impossible to satisfy it for this particular case. (Note, `let<space>` and `var<space>` are exactly 4 characters long so they don't suffer from this issue.)

**Comment by vitorbal** [comment]
Hi @jonathanj, thanks for reaching out! There's been some discussions around this already in https://github.com/eslint/eslint/issues/6296.

Basically, the indent rule is pretty complicated right now and we would like to sketch a proposal on how to re-work some of its options, including `VariableDeclarator`. If you have any suggestions, be sure to drop a comment on that discussion.

Closing this in favor of #6296, please re-open if anyone disagrees.
